lib/src/layer/polyline_layer/polyline_layer.dart

```diff
@@ -146,17 +165,59 @@ class _PolylineLayerState<R extends Object> extends State<PolylineLayer<R>>
     final (xEast, _) = projection.projectXY(const LatLng(0, 180));
     for (final projectedPolyline in polylines) {
       final polyline = projectedPolyline.polyline;
+      final boundingBox = polyline.boundingBox;
+
+      /// Check if the camera and the polyline overlap, latitude-wise.
+      bool isOverlappingLatitude() {
+        if (boundsAdjusted.north < boundingBox.south) {
+          return false;
+        }
+        if (boundsAdjusted.south > boundingBox.north) {
+          return false;
+        }
+        return true;
+      }
+
+      /// Check if the camera and the polyline overlap, longitude-wise.
+      bool isOverlappingLongitude() {
+        if (boundsAdjusted.east < boundingBox.west) {
+          return false;
+        }
+        if (boundsAdjusted.west > boundingBox.east) {
+          return false;
+        }
+        return true;
+      }
+
+      /// Check if the camera longitude bounds are reliable, world-wise.
+      bool areLongitudeBoundsReliable() {
+        if (boundsAdjusted.east == LatLngBounds.maxLongitude) {
+          return false;
+        }
+        if (boundsAdjusted.west == LatLngBounds.minLongitude) {
+          return false;
+        }
+        return true;
+      }
 
       // Test bounding boxes to avoid potentially expensive aggressive culling
       // when none of the line is visible
-      if (!boundsAdjusted.isOverlapping(polyline.boundingBox)) continue;
+      // First check, bullet-proof, focusing on latitudes.
+      if (!isOverlappingLatitude()) continue;
 
-      // Gradient poylines cannot be easily segmented
+      // Gradient polylines cannot be easily segmented
       if (polyline.gradientColors != null) {
         yield projectedPolyline;
         continue;
       }
 
+      // TODO: think about how to cull polylines in a forced "drawInSingleWorld".
+      // As what may be culled in a world may not be culled in a next world.
+      if (widget.drawInSingleWorld) {
+        yield projectedPolyline;
+        continue;
+      }
+
```

Comment:
Is this culling something that could be more easily implemented if we have a method - or are we limited by the tools we have again. For example, I realise `LatLngBounds` can't really support what we're doing, so could something like that be an issue?

Reply:
Ok, I've removed this check then. I think we're ready to go with this hopefully :)